packages/mui-base/src/SliderUnstyled/SliderUnstyled.test.tsx

```diff
@@ -136,31 +152,42 @@ describe('<SliderUnstyled />', () => {
     });
   });
 
-  [
+  type Values = Array<[string, number[]]>;
+
+  const values = [
     ['readonly range', Object.freeze([2, 1])],
     ['range', [2, 1]],
-  ].forEach(([valueLabel, value]) => {
+  ] as Values;
+  values.forEach(([valueLabel, value]) => {
     it(`calls onChange even if the ${valueLabel} did not change`, () => {
       const handleChange = spy();
+
       const { container } = render(
         <SliderUnstyled min={0} max={5} onChange={handleChange} value={value} />,
       );
-      stub(container.firstChild, 'getBoundingClientRect').callsFake(() => ({
-        width: 100,
-        height: 10,
-        bottom: 10,
-        left: 0,
-      }));
+
+      if (container.firstChild !== null) {
+        // @ts-expect-error - container.firstChild doesn't have getBoundingClientRect method.
+        stub(container.firstChild, 'getBoundingClientRect').callsFake(() => ({
```

Comment:
I wonder if we can change it to as shown below. This way we can remove `@ts-expect-error` comment. The `firstChild` node does not contain `getBoundingClientRect` method. We can instead select the `firstChild` which is the slider root by passing `data-testid` to slider and using `getByTestId`. That way the root element has the `getBoundingClientRect` method. cc: @mnajdova @michaldudak 
```diff
--- a/packages/mui-base/src/SliderUnstyled/SliderUnstyled.test.tsx
+++ b/packages/mui-base/src/SliderUnstyled/SliderUnstyled.test.tsx
@@ -162,19 +162,29 @@ describe('<SliderUnstyled />', () => {
     it(`calls onChange even if the ${valueLabel} did not change`, () => {
       const handleChange = spy();

-      const { container } = render(
-        <SliderUnstyled min={0} max={5} onChange={handleChange} value={value} />,
+      const { getByTestId } = render(
+        <SliderUnstyled
+          min={0}
+          max={5}
+          onChange={handleChange}
+          value={value}
+          data-testid="slider-root"
+        />,
       );

-      if (container.firstChild !== null) {
-        // @ts-expect-error - container.firstChild doesn't have getBoundingClientRect method.
-        stub(container.firstChild, 'getBoundingClientRect').callsFake(() => ({
-          width: 100,
-          height: 10,
-          bottom: 10,
-          left: 0,
-        }));
-      }
+      const sliderRoot = getByTestId('slider-root');
+
+      stub(sliderRoot, 'getBoundingClientRect').callsFake(() => ({
+        width: 100,
+        height: 10,
+        bottom: 10,
+        left: 0,
+        x: 0,
+        y: 0,
+        right: 0,
+        top: 0,
+        toJSON() {},
+      }));

       // pixel:  0   20  40  60  80  100
       // slider: |---|---|---|---|---|
@@ -182,12 +192,10 @@ describe('<SliderUnstyled />', () => {
       // value:      ↑   ↑
       // mouse:           ↑

-      if (container.firstChild !== null) {
-        fireEvent.mouseDown(container.firstChild, {
-          buttons: 1,
-          clientX: 41,
-        });
-      }
+      fireEvent.mouseDown(sliderRoot, {
+        buttons: 1,
+        clientX: 41,
+      });

       expect(handleChange.callCount).to.equal(1);
       expect(handleChange.args[0][1]).not.to.equal(value);
```

@sai6855 Don't commit until it is confirmed.

Reply:
Sorry for the delay, @ZeeshanTamboli please feel free to apply the change and merge the PR.